pkg/hintrunner/zero/zerohint_math.go

```diff
@@ -505,6 +505,59 @@ func createSplitIntHinter(resolver hintReferenceResolver) (hinter.Hinter, error)
 	return newSplitIntHint(output, value, base, bound), nil
 }
 
+func newPowHint(locs, prevLocs hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "Pow",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> ids.locs.bit = (ids.prev_locs.exp % PRIME) & 1
+			/*> struct LoopLocals {
+				bit: felt,
+				temp0: felt,
+
+				res: felt,
+				base: felt,
+				exp: felt,
+			} */
+			const expStructOffset = 4
+			locsBitAddress, err := locs.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+			prevLocsBitAddress, err := prevLocs.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+			loopLocals, err := hinter.GetConsecutiveValues(vm, prevLocsBitAddress, expStructOffset+1)
+			if err != nil {
+				return err
+			}
+			prevLocsExp, err := loopLocals[expStructOffset].FieldElement()
+			if err != nil {
+				return err
+			}
+			var prevLocsExpBig big.Int
+			prevLocsExp.BigInt(&prevLocsExpBig)
+			locsBitBig := new(big.Int).And(&prevLocsExpBig, big.NewInt(1))
+			v := memory.MemoryValueFromFieldElement(new(fp.Element).SetBigInt(locsBitBig))
+			return vm.Memory.WriteToAddress(&locsBitAddress, &v)
+		},
+	}
+}
+
+func createPowHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
+	locs, err := resolver.GetCellRefer("locs")
+	if err != nil {
+		return nil, err
+	}
+	prev_locs, err := resolver.GetCellRefer("prev_locs")
+	if err != nil {
+		return nil, err
+	}
+	locsRes := hinter.Deref{Deref: locs}
+	prevLocsRes := hinter.Deref{Deref: prev_locs}
+	return newPowHint(locsRes, prevLocsRes), nil
```

Comment:
This part I don't like so much. The `newPowHint` method does not have to comply with a specific interface so in reality you could pass whichever parameter types you need. The problem is with the testing interface, right? Because we create only `Deref` structs for the operanders.
I would prefer a solution where the testing suit allows us to pass the types we need, we shouldn't have to wrap the operand just to unwrap it later. Also our decisions regarding implementation shouldn't be determined by the way we do testing, but the other way around.
However, this would imply we have to modify the current testing structure, so let's leave this as it is and implement a solution in another PR.

Just created an issue for it https://github.com/NethermindEth/cairo-vm-go/issues/332